compiler/rustc_privacy/src/lib.rs

```diff
@@ -433,7 +433,12 @@ impl<'tcx> EmbargoVisitor<'tcx> {
         let old_level = self.get(def_id);
         // Accessibility levels can only grow.
         if level > old_level {
-            self.access_levels.set_access_level(def_id, level.unwrap());
+            let module_id = self.tcx.parent_module_from_def_id(def_id);
+            self.access_levels.set_access_level(
+                def_id,
+                ty::Visibility::Restricted(module_id),
```

Comment:
Possible perf improvement: make `fn set_access_level` and `fn update` take `impl FnOnce() -> Visibility` instead of just `Visibility` to avoid calculating parent modules unnecessarily.